src/emcee/ensemble.py

```diff
@@ -178,6 +191,31 @@ def random_state(self, state):
         except:
             pass
 
+    def _check_random_state(self, seed, state):
```

Comment:
`_check_random_state` should probably be a standalone function:

```
    def _check_random_state(seed, state=None):
        """Check seed argument and set RandomState.
        Based on scikit-learn utils/validation.py.
        """
        if isinstance(seed, (int, np.integer)) or seed is None:
            _random = np.random.mtrand.RandomState()
            if state is not None:
                _random.set_state(state)
            if seed is not None:
                _random.seed(seed)
            return _random
        elif isinstance(seed, np.random.RandomState):
            _random = seed
            if state is not None:
                _random.set_state(state)
            return _random
        else:
            try:
            # Generator is only available in numpy >= 1.17
                if isinstance(seed, np.random.Generator):
                    return seed
            except AttributeError:
                pass
            raise TypeError(
                "seed must be an int, np.random.RandomState, np.random.Generator or "
                "None of type {}".format(type(seed))
            )
```

Reply:
I think it's only called from the `random_state` property setter, nor does it access much object state, so there's no real need for it to be an instance method.